google/cloud/opentelemetry/internal/monitored_resource.cc

```diff
@@ -29,15 +32,25 @@ namespace {
 
 namespace sc = opentelemetry::sdk::resource::SemanticConventions;
 
-std::string AsString(
-    opentelemetry::sdk::common::OwnedAttributeValue attribute) {
-  if (absl::holds_alternative<std::string>(attribute)) {
-    return absl::get<std::string>(attribute);
+struct AsStringVisitor {
+  template <typename T>
+  std::string operator()(std::vector<T> const& v) const {
+    return absl::StrCat("[", absl::StrJoin(v, ", "), "]");
```

Comment:
Consider whether the vector versions should call through to the scalar versions.